keras_cv/models/classification/image_classifier.py

```diff
@@ -0,0 +1,188 @@
+    backbone_presets_with_weights,
+)
+from keras_cv.models.classification.image_classifier_presets import (
+    classifier_presets,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class PooledDenseHead(layers.Layer):
+    """General purpose head for image classification.
+
+    Applies GlobalAveragePooling followed by a Dense layer mapping to the
+    number of classes to the Backbone output.
+
+    Args:
+        num_classes: int, number of classes to predict.
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+    """
+
+    def __init__(
+        self,
+        num_classes=2,
+        activation=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.pooling = layers.GlobalAveragePooling2D(name="avg_pool")
+        self.dense = layers.Dense(
+            num_classes,
+            activation=activation,
+            name="predictions",
+        )
+        self.num_classes = num_classes
+        self.activation = activation
+
+    def call(self, inputs):
+        x = self.pooling(inputs)
+        return self.dense(x)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "num_classes": self.num_classes,
+                "activation": self.activation,
+            }
+        )
+        return config
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ImageClassifier(Task):
+    """Image classifier with configurable prediction head.
+
+    Args:
+        backbone: `keras.Model` instance, the backbone architecture of the 
+            classifier called on the inputs.
+        num_classes: int, number of classes to predict.
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+        head: keras.layers.Layer. Prediction head for classifier. Must take
+            arguments "num_classes" and "activation" and have no other
+            positional arguments.
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained backbone
+    model = keras_cv.models.ImageClassifier.from_preset(
+        "resnet50_v2_imagenet",
+        num_classes=4,
+    )
+    output = model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = keras_cv.models.ImageClassifier(
+        backbone=keras_cv.models.ResNet50V2Backbone(),
+        num_classes=4,
+        head=keras_cv.models.classification.PooledDenseHead,
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        backbone,
+        num_classes=2,
```

Comment:
`num_classes` should be required

Reply:
Is this a strong take? Both HF and KerasNLP set two as the default so there would be some standardization

Reply:
Personally feel pretty strongly that it should be required in CV.

Just curious the history --- Why is it two in NLP/HF?  Toxic/non-toxic being the most common use case or something of this sort?

Reply:
Strong take IMO, there is no right default value. Having 2 classes *and* softmax activation as the default would especially be an antipattern since you'd use a binary classifier in the 2-class case.